packages/react-aria-components/src/Tree.tsx

```diff
@@ -0,0 +1,548 @@
+  let [expandedKeys, setExpandedKeys] = useControlledState(
+    propExpandedKeys ? convertExpanded(propExpandedKeys) : undefined,
+    propDefaultExpandedKeys ? convertExpanded(propDefaultExpandedKeys) : new Set(),
+    onExpandedChange
+  );
+
+  let flattenedCollection = useMemo(() => {
+    return new TreeCollection<object>({collection, expandedKeys});
+  }, [collection, expandedKeys]);
+
+  let state = useTreeState({
+    ...props,
+    selectionMode,
+    expandedKeys,
+    onExpandedChange: setExpandedKeys,
+    collection: flattenedCollection,
+    children: undefined,
+    disabledBehavior
+  });
+
+  let {gridProps} = useTreeGridList(props, state, ref);
+  let children = useCachedChildren({
+    items: state.collection as ICollection<Node<T>>,
+    children: (item: Node<T>) => {
+      switch (item.type) {
+        case 'item':
+          return <TreeRow item={item} />;
+        default:
+          throw new Error('Unsupported node type in Tree: ' + item.type);
+      }
+    }
+  });
+
+  let {focusProps, isFocused, isFocusVisible} = useFocusRing();
+  let renderValues = {
+    isEmpty: state.collection.size === 0,
+    isFocused,
+    isFocusVisible,
+    state
+  };
+
+  let renderProps = useRenderProps({
+    className: props.className,
+    style: props.style,
+    defaultClassName: 'react-aria-Tree',
+    values: renderValues
+  });
+
+  let emptyState: ReactNode = null;
+  let emptyStatePropOverrides: HTMLAttributes<HTMLElement> | null = null;
+  if (state.collection.size === 0 && props.renderEmptyState) {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    let {isEmpty, ...values} = renderValues;
+    let content = props.renderEmptyState({...values});
+    let treeGridRowProps = {
+      'aria-level': 1,
+      'aria-posinset': 1,
+      'aria-setsize': 1
+    };
+
+    emptyState = (
+      <div role="row" style={{display: 'contents'}} {...treeGridRowProps}>
+        <div role="gridcell" style={{display: 'contents'}}>
+          {content}
+        </div>
+      </div>
+    );
+  }
+
+  return (
+    <FocusScope>
+      <div
+        {...filterDOMProps(props)}
+        {...renderProps}
+        {...mergeProps(gridProps, focusProps, emptyStatePropOverrides)}
+        ref={ref}
+        slot={props.slot || undefined}
+        onScroll={props.onScroll}
+        data-empty={state.collection.size === 0 || undefined}
+        data-focused={isFocused || undefined}
+        data-focus-visible={isFocusVisible || undefined}>
+        <Provider
+          values={[
+            [TreeStateContext, state]
+          ]}>
+          {children}
+        </Provider>
+        {emptyState}
+      </div>
+    </FocusScope>
+  );
+}
+
+/**
+ * A tree provides users with a way to navigate nested hierarchical information, with support for keyboard navigation
+ * and selection.
+ */
+const _Tree = /*#__PURE__*/ (forwardRef as forwardRefType)(Tree);
+export {_Tree as Tree};
```

Comment:
Should this be called TreeView to better match the ARIA APG https://www.w3.org/WAI/ARIA/apg/patterns/treeview/

Reply:
happy to change it, mainly was concerned about if the RSP tree would be called TreeView too haha. Anyone else have any thoughts? Maybe TreeGrid (except it is only one column...)

Reply:
I think it's fine if they have the same name. IMO: TreeView > Tree > TreeGrid